google-cloud-spanner/src/main/java/com/google/cloud/spanner/Options.java

```diff
@@ -131,7 +135,29 @@ public interface UpdateAdminApiOption extends AdminApiOption {}
   public interface QueryOption {}
 
   /** Marker interface to mark options applicable to write operations */
-  public interface TransactionOption {}
+  public interface TransactionOption {
+    Predicate<TransactionOption> isValidIsolationLevelOption =
+        txnOption ->
+            txnOption instanceof RepeatableReadOption || txnOption instanceof SerializableOption;
+
+    /**
+     * Combines two arrays of TransactionOption, with primaryOptions taking precedence in case of
+     * conflicts. Note that {@link
+     * com.google.cloud.spanner.SpannerOptions.Builder.TransactionOptions} supports only the {@link
+     * IsolationLevel} TransactionOption, meaning spannerOptions will contain a maximum of one
+     * TransactionOption.
+     */
+    static TransactionOption[] combine(
+        TransactionOption[] primaryOptions, TransactionOption[] spannerOptions) {
+      if (spannerOptions == null
+          || Arrays.stream(primaryOptions).anyMatch(isValidIsolationLevelOption)) {
+        return primaryOptions;
+      } else {
+        return Stream.concat(Arrays.stream(primaryOptions), Arrays.stream(spannerOptions))
+            .toArray(TransactionOption[]::new);
+      }
+    }
+  }
```

Comment:
I think that this block of code could be significantly simplified by just moving the check to the place where we are actually building the request, and then do something like this there:

```java
IsolationLevel isolationLevel = MoreObjects.firstNonNull(
    options.getIsolationLevel(),
    spannerOptions.getDefaultIsolationLevel());
if (isolationLevel != null) {
  transactionOptions.setIsolationLevel(isolationLevel);
}
```